cookbooks/solr/recipes/default.rb

```diff
@@ -0,0 +1,140 @@
+#
+# Cookbook Name:: solr
+# Recipe:: default
+#
+# We specify what version we want below.
+use_default_java = false
+java_package_name = "dev-java/icedtea-bin"
+java_version = node['solr']['java_version']
+java_eselect_version = node['solr']['java_eselect_version']
+solr_version = node['solr']['solr_version']
+solr_file = "solr-#{solr_version}.tgz"
+solr_url = "https://archive.apache.org/dist/lucene/solr/#{solr_version}/#{solr_file}"
+core_name = node['solr']['core_name']
+
+username = node['dna']['users'].first['username']
+
+# Install Solr
+if ('solo' == node['dna']['instance_role'])  ||
```

Comment:
@crigor's approach with https://github.com/engineyard/ey-cookbooks-stable-v5/blob/24b08942a8d295e229037af63789582b2f0df17d/cookbooks/delayed_job4/attributes/default.rb#L24 looks good.

Just need to confirm and test the case for customization.